pkg/controller/cache/redis.go

```diff
@@ -18,297 +18,159 @@ package cache
-	if _, err := a.client.Create(ctx, r.Spec.ResourceGroupName, n, redis.NewCreateParameters(r)); err != nil {
-		r.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
-		return true
-	}
-
-	r.Status.ResourceName = n
-	meta.AddFinalizer(r, finalizerName)
-	r.Status.SetConditions(runtimev1alpha1.ReconcileSuccess())
-	return true
-}
-
-func (a *azureRedisCache) Sync(ctx context.Context, r *v1alpha3.Redis) bool {
-	n := redis.NewResourceName(r)
-	cacheResource, err := a.client.Get(ctx, r.Spec.ResourceGroupName, n)
-	if err != nil {
-		r.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
-		return true
-	}
-
-	r.Status.State = string(cacheResource.ProvisioningState)
-
-	switch r.Status.State {
-	case v1alpha3.ProvisioningStateSucceeded:
-		// TODO(negz): Set r.Status.State to something like 'Ready'? The Azure
-		// portal shows an instance as 'Ready', but the API shows only that the
-		// provisioning state is 'Succeeded'. It's a little weird to see a Redis
-		// resource in state 'Succeeded' in kubectl.
-		r.Status.SetConditions(runtimev1alpha1.Available())
-		if r.Status.Endpoint != "" {
-			resource.SetBindable(r)
-		}
-	case v1alpha3.ProvisioningStateCreating:
-		r.Status.SetConditions(runtimev1alpha1.Creating(), runtimev1alpha1.ReconcileSuccess())
-		return true
-	case v1alpha3.ProvisioningStateDeleting:
-		r.Status.SetConditions(runtimev1alpha1.Deleting(), runtimev1alpha1.ReconcileSuccess())
-		return false
-	default:
-		// TODO(negz): Don't requeue in this scenario? The instance could be
-		// failed, disabled, scaling, etc.
-		r.Status.SetConditions(runtimev1alpha1.ReconcileSuccess())
-		return true
-	}
-
-	r.Status.Endpoint = azure.ToString(cacheResource.HostName)
-	r.Status.Port = azure.ToInt(cacheResource.Port)
-	r.Status.SSLPort = azure.ToInt(cacheResource.SslPort)
-	r.Status.RedisVersion = azure.ToString(cacheResource.RedisVersion)
-	r.Status.ProviderID = azure.ToString(cacheResource.ID)
-
-	if !redis.NeedsUpdate(r, cacheResource) {
-		r.Status.SetConditions(runtimev1alpha1.ReconcileSuccess())
-		return false
-	}
+// SetupWithManager creates a new MySQLServer Controller and adds it to the
+// Manager with default RBAC. The Manager will set fields on the Controller and
+// start it when the Manager is Started.
+func (c *Controller) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(v1alpha3.RedisClassGroupVersionKind),
+		resource.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: redis.NewClient}))
 
-	if _, err := a.client.Update(ctx, r.Spec.ResourceGroupName, n, redis.NewUpdateParameters(r)); err != nil {
-		r.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
-		return true
-	}
+	name := strings.ToLower(fmt.Sprintf("%s.%s", v1alpha3.RedisKind, v1alpha3.Group))
 
-	r.Status.SetConditions(runtimev1alpha1.ReconcileSuccess())
-	return false
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha3.Redis{}).
```

Comment:
Will we bump to v1beta1  in a future commit?

Reply:
yes, done.